sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_common/_base_regex_eval.py

```diff
@@ -0,0 +1,183 @@
+        :type eval_input: Dict
+        :return: The evaluation result.
+        :rtype: Dict
+        """
+        response = eval_input["response"]
+        ground_truth = eval_input["ground_truth"]
+
+        result = self.update(
+                prediction=response,
+                label=ground_truth,
+                json_data={}
+            )
+
+        return result
+
+    def update(self, prediction: str, label: str, json_data: dict) -> Dict[str, Any]:
+        if self.is_missing_regex_patterns:
+            self.regex_patterns = self.get_regex_patterns(prediction, label, json_data)
+        expected_answer = self.extract_expected_answer(label, json_data)
+        regex_match = self.extract_regex(prediction, label, json_data)
+
+        if regex_match:
+            extracted_answer = regex_match.group(1).strip()
+            follow_instruction = 1
+            chain_of_thought_length = self._get_chain_of_thought_length(
+                prediction, regex_match.start()
+            )
+            self.chain_of_thought_lengths.append(chain_of_thought_length)
+        else:
+            extracted_answer = ""
+            follow_instruction = 0
+            chain_of_thought_length = -1
+
+        score = self.compute_match(expected_answer, extracted_answer, label, json_data)
+        self.scores.append(score)
+        self.follow_instructions.append(follow_instruction)
+
+        return {
+            "score": score,
+            "follow_instruction": follow_instruction,
+            "extracted_answer": extracted_answer,
+            "chain_of_thought_length": chain_of_thought_length,
+        }
+
+    def __aggregate__(self, line_results: List[str]) -> Dict[str, float]:
+        """
+        Base aggregation method for the RegexEvaluatorBase.
+        This method aggregates the results of the metric across multiple lines of input data.
+        Throws an exception if the input list or chain lengths is empty.
+        """
+        if not line_results:
+            raise ValueError("line_results is empty passed to __aggregate__")
+
+        # collect individual metric values
+        scores = [r["score"] for r in line_results]
+        follow_flags = [r["follow_instruction"] for r in line_results]
+
+        # only include chain lengths where instruction was followed (non-negative)
+        chain_lengths = [
+            r.get("chain_of_thought_length", -1)
+            for r in line_results
+            if r.get("chain_of_thought_length", -1) >= 0
+        ]
+
+        # compute aggregate metrics
+        accuracy = np.mean(scores)
+        follow_instruction_rate = np.mean(follow_flags)
+        chain_of_thought_length = np.mean(chain_lengths) if chain_lengths else -1
+
+        return {
+            "accuracy": accuracy,
+            "follow_instruction_rate": follow_instruction_rate,
+            "chain_of_thought_length": chain_of_thought_length,
+        }
+
+    def get_regex_patterns(
```

Comment:
Please start name of method with underscore if it is not be exposed to customers